frontend/rust-lib/flowy-database2/src/services/group/configuration.rs

```diff
@@ -474,25 +474,32 @@ fn merge_groups(
     .into_iter()
     .map(|group| (group.id.clone(), group))
     .collect();
+  let mut no_status_group_inserted = false;
 
   // The group is ordered in old groups. Add them before adding the new groups
   for old in old_groups {
     if let Some(new) = new_group_map.shift_remove(&old.id) {
       merge_result.all_groups.push(new.clone());
+    } else if matches!(&no_status_group, Some(group) if group.id == old.id) {
+      merge_result
+        .all_groups
+        .push(no_status_group.clone().unwrap());
+      no_status_group_inserted = true;
     } else {
       merge_result.deleted_groups.push(old);
     }
   }
 
   // Find out the new groups
-  merge_result
-    .all_groups
-    .extend(new_group_map.values().cloned());
-  merge_result.new_groups.extend(new_group_map.into_values());
+  let new_groups = new_group_map.into_values();
+  for (_, group) in new_groups.into_iter().enumerate() {
+    merge_result.all_groups.push(group.clone());
+    merge_result.new_groups.push(group);
+  }
```

Comment:
Oops. I think this part can be reverted? Otherwise, this looks good to me

Reply:
Done.